addons/arsenal/functions/fnc_updateUniqueItemsList.sqf

```diff
@@ -0,0 +1,118 @@
+/*
+ * Author: Alganthe
+ * Update the list of unique items.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+*/
+#include "script_component.hpp"
+#include "..\defines.hpp"
+
+GVAR(virtualItems) set [18, []];
+GVAR(virtualItems) set [19, []];
+GVAR(virtualItems) set [20, []];
+GVAR(virtualItems) set [21, []];
+GVAR(virtualItems) set [22, [[], [], [], []]];
+GVAR(virtualItems) set [23, []];
+GVAR(virtualItems) set [24, []];
+
+private _array = LIST_DEFAULTS select 2;
+private _itemsCache = uiNamespace getVariable QGVAR(configItems);
+
+private _configCfgWeapons = configFile >> "CfgWeapons";
+private _configMagazines = configFile >> "CfgMagazines";
+private _configVehicles = configFile >> "CfgVehicles";
+private _configGlasses = configFile >> "CfgGlasses";
+
+{
+    switch true do {
+        // Weapon mag
+        case (
+                isClass (_configMagazines >> _x) && 
```

Comment:
again you might be able to move the repeated isClass check outside the case statements and only check it once.
You can also store result in a variable before the switch statement.